exetera/core/utils.py

```diff
@@ -393,4 +393,37 @@ def one_dim_data_to_indexed_for_test(data, field_size):
                 length += 1
         indices[0, i + 1] = indices[0, i] + length
          
-    return indices, values, offsets, count_row
\ No newline at end of file
+    return indices, values, offsets, count_row
+
+
+def to_timestamp(dt):
+    """
+    This is an alternative of datetime.timestamp() as such function will raise an OSError on windoes if year is less
+    than 1970 or greater than 3002.
+
+    :param dt: The datetime instance to convert.
+
+    :return: The timestamp of the date.
+    """
+    if not isinstance(dt, datetime):
+        raise TypeError("Please use a datetime variable as argument.")
+    DATE_TIME_EPOCH = datetime(1970, 1, 1, tzinfo=timezone.utc)
```

Comment:
This constant should be in the global scope at the top of this file, and removed from `to_datetime` as well.